snmp/datadog_checks/snmp/cmd.py

```diff
@@ -0,0 +1,134 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from pyasn1.type.univ import Null
+from pysnmp.entity.rfc3413 import cmdgen
+from pysnmp.hlapi.asyncore.cmdgen import vbProcessor
+from pysnmp.proto import errind
+from pysnmp.proto.rfc1905 import endOfMibView
+
+from datadog_checks.base.errors import CheckException
+
+
+def _handle_error(ctx, config):
+    error = ctx['error']
+    if error:
+        message = '{} for instance {}'.format(error, config.ip_address)
+        raise CheckException(message)
+
+
+def snmp_get(config, oids, lookup_mib):
+    """Call SNMP GET on a list of oids."""
+
+    def callback(snmpEngine, sendRequestHandle, errorIndication, errorStatus, errorIndex, varBinds, cbCtx):
+        var_binds = vbProcessor.unmakeVarBinds(snmpEngine, varBinds, lookup_mib)
+
+        cbCtx['error'] = errorIndication
```

Comment:
Should we also store/check for `errorStatus`? (I remember `errorIndication` is for "server-side" errors, eg. server timeouts, while `errorStatus` is for "client-side" errors, eg. invalid auth.)

Reply:
I spent some time looking at it, and it seems that errorStatus are unused. When you have invalid authentication, the server actually doesn't respond.

It'd be nice to add more test for it, of course.